validator/src/main/java/com/google/daq/mqtt/registrar/LocalDevice.java

```diff
@@ -309,12 +307,17 @@ private DeviceCredential getDeviceCredential(String keyFile) throws IOException
   }
 
   private Set<String> keyFiles() {
-    if (!isDirectConnect()) {
+    if (!isDirectConnect() || !hasAuthType()) {
       return ImmutableSet.of();
     }
+    String authType = getAuthType();
+    Set<String> certFile = (authType.equals(ES_CERT_TYPE) || authType.equals(RSA_CERT_TYPE)) ? Set.of(certFile()) : null;
```

Comment:
I would move all of this logic into the certFile() function (and rename it getCertSet()) -- no need to have this exposed here since otherwise that function is trivial.

Also, variable should be named certFiles or certSet -- it's not a singular file.

Reply:
In the tests I've done with the RS256_X509 and ES256_X509 cases I've only experienced having a single cert file, so I've modelled this function similarly to the public key one.
Does it mean I need to change both the cert and the public key functions?